docs/samples/client/oauth/csharp/UnbrandedCredential.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Demonstrates creating a client with an unbranded authentication token provider.
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+
+namespace UnbrandedTypeSpec
+{
+    /// <summary>
+    /// Demonstrates creating a client with an unbranded authentication token provider.
+    /// </summary>
+    public partial class UnbrandedTypeSpecClient
+    {
+        private readonly Dictionary<string, object>[] flows = [
+            new Dictionary<string, object> {
+                    { GetTokenOptions.ScopesPropertyName, new string[] { "baselineScope" } },
+                    { GetTokenOptions.TokenUrlPropertyName , "https://myauthserver.com/token"},
+                    { GetTokenOptions.RefreshUrlPropertyName, "https://myauthserver.com/refresh"}
+                }
+        ];
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnbrandedTypeSpecClient"/> class.
+        /// </summary>
+        /// <param name="uri">The URI of the service.</param>
+        /// <param name="credential">The authentication token provider.</param>
+        public UnbrandedTypeSpecClient(Uri uri, AuthenticationTokenProvider credential)
```

Comment:
Are we adding a sample implementation of the token provider also?

Reply:
Do you mean the ApiKeyCredential? We have an implementation here https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs#L37 :)

Reply:
no - like [this one](https://github.com/azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.Experimental/tests/Auth/TokenProviderTests.cs#L103)

Reply:
No, we aren't adding sample implementations as part of this work. This should be handled by the Identity feature crew.